templates/author-json.php

```diff
@@ -1,10 +1,6 @@
 <?php
 $user = \Activitypub\Collection\Users::get_by_id( \get_the_author_meta( 'ID' ) );
 
-$user->set_context(
```

Comment:
I used that to set it only where needed. With your changes, we have the context also on the items in the collections, for example in the outbox and in sub-objects like the `Object` attribute. But we only want the context on the top level.

In the following example we have 3 different context definitions for example:

```JSON
{
    "@context": [
        "https:\\/\\/www.w3.org\\/ns\\/activitystreams"
    ],
    "id": "http:\\/\\/localhost:8076\\/wp-json\\/activitypub\\/1.0\\/users\\/1\\/outbox",
    "generator": "http:\\/\\/wordpress.org\\/?v=6.4.2",
    "actor": "http:\\/\\/localhost:8076\\/author\\/test\\/",
    "type": "OrderedCollectionPage",
    "partOf": "http:\\/\\/localhost:8076\\/wp-json\\/activitypub\\/1.0\\/users\\/1\\/outbox",
    "totalItems": 15,
    "first": "http:\\/\\/localhost:8076\\/wp-json\\/activitypub\\/1.0\\/users\\/1\\/outbox?page=1",
    "last": "http:\\/\\/localhost:8076\\/wp-json\\/activitypub\\/1.0\\/users\\/1\\/outbox?page=2",
    "next": "http:\\/\\/localhost:8076\\/wp-json\\/activitypub\\/1.0\\/users\\/1\\/outbox?page=2",
    "orderedItems": [
        {
            "@context": [],
            "type": "Create",
            "object": {
                "@context": [
                    "https:\\/\\/www.w3.org\\/ns\\/activitystreams",
                    {
                        "Hashtag": "as:Hashtag"
                    }
                ],
                "id": "http:\\/\\/localhost:8076\\/2023\\/12\\/26\\/classic-editor-images\\/",
                "type": "Note",
                "attachment": [
                    {
                        "type": "Image",
                        "url": "http:\\/\\/localhost:8076\\/wp-content\\/uploads\\/2023\\/12\\/337b3d7a89119c55a1a4c13bf5dd9332.jpeg",
                        "mediaType": "image\\/jpeg"
                    },
                    {
                        "type": "Image",
                        "url": "http:\\/\\/localhost:8076\\/wp-content\\/uploads\\/2023\\/10\\/273020110-810ab794-55f9-4a71-999f-0463921219d2.jpg",
                        "mediaType": "image\\/jpeg"
                    }
                ],
                "attributedTo": "http:\\/\\/localhost:8076\\/author\\/test\\/",
                "content": "<p>&nbsp;<br \\/>Hallo<br \\/>&nbsp;<\\/p><p><a href=\"http:\\/\\/localhost:8076\\/2023\\/12\\/26\\/classic-editor-images\\/\">http:\\/\\/localhost:8076\\/2023\\/12\\/26\\/classic-editor-images\\/<\\/a><\\/p>",
                "contentMap": {
                    "en": "<p>&nbsp;<br \\/>Hallo<br \\/>&nbsp;<\\/p><p><a href=\"http:\\/\\/localhost:8076\\/2023\\/12\\/26\\/classic-editor-images\\/\">http:\\/\\/localhost:8076\\/2023\\/12\\/26\\/classic-editor-images\\/<\\/a><\\/p>"
                },
                "published": "2023-12-26T20:04:02Z",
                "tag": [],
                "url": "http:\\/\\/localhost:8076\\/2023\\/12\\/26\\/classic-editor-images\\/",
                "to": [
                    "https:\\/\\/www.w3.org\\/ns\\/activitystreams#Public",
                    "http:\\/\\/localhost:8076\\/wp-json\\/activitypub\\/1.0\\/users\\/1\\/followers"
                ],
                "cc": []
            },
            "actor": "http:\\/\\/localhost:8076\\/author\\/test\\/",
            "id": "http:\\/\\/localhost:8076\\/2023\\/12\\/26\\/classic-editor-images\\/#activity",
            "published": "2023-12-26T20:04:02Z",
            "to": [
                "https:\\/\\/www.w3.org\\/ns\\/activitystreams#Public",
                "http:\\/\\/localhost:8076\\/wp-json\\/activitypub\\/1.0\\/users\\/1\\/followers"
            ],
            "cc": []
        }
    ]
}
```

Also: If you change it to have different and extendable contexts, we need a way to merge sub contexts (in collections or sub-objects) into the main.

Reply:
I had a dream of really getting the a compacted JsonLD context from the objects and their attributes, but its anything but trivial. I gave this idea up, in the end nobody parses it yet.

So right now am fine with the most minimal hard-coded approach as long it does not require to use the context filter for objects like Event. 

And if at the moment we need to change the JsonLD context we can still filter it, or wait for a proper solution until it becomes relevant. If one wants to get things right this would be needed for the address of Events: some send a xsd:Text, other send a PostalAddress (schema.org).

Reply:
And thanks for pointing out those other use cases: I did not have these cases on my radar.